src/test/javascript/spec/component/course/course-group.component.spec.ts

```diff
@@ -259,4 +268,24 @@ describe('Course Management Detail Component', () => {
             expect(comp.searchTextFromUser(user)).toBe('');
         });
     });
+
+    it('should generate csv correctly', () => {
+        comp.allCourseGroupUsers = [courseGroupUser, courseGroupUser2];
+        comp.courseGroup = CourseGroup.STUDENTS;
+        comp.course = course;
+        const exportAsCsvMock = jest.spyOn(comp, 'exportAsCsv');
+
+        comp.exportUserInformation();
+
+        expect(exportAsCsvMock).toHaveBeenCalledTimes(1);
+        const generatedRows = exportAsCsvMock.mock.calls[0][0];
+
+        for (let index = 0; index < generatedRows.length; index++) {
+            const userRow = generatedRows[index];
+            expect(userRow[NAME_KEY]).toBe(comp.allCourseGroupUsers[index].name ?? '');
+            expect(userRow[USERNAME_KEY]).toBe(comp.allCourseGroupUsers[index].login ?? '');
+            expect(userRow[EMAIL_KEY]).toBe(comp.allCourseGroupUsers[index].email ?? '');
+            expect(userRow[REGISTRATION_NUMBER_KEY]).toBe(comp.allCourseGroupUsers[index].visibleRegistrationNumber ?? '');
+        }
```

Comment:
Please avoid replicating the functionality of the code in the test. Instead you could expect constant values with something like
```ts
const expectedRow1 = {};
expectedRow1[NAME_KEY] = '';
expectedRow1[USERNAME_KEY] = courseGroupUser.login;
expectedRow1[EMAIL_KEY] = '';
expectedRow1[REGISTRATION_NUMBER_KEY] = '';

const expectedRow2 = {};
expectedRow2[NAME_KEY] = '';
expectedRow2[USERNAME_KEY] = courseGroupUser2.login;
expectedRow2[EMAIL_KEY] = '';
expectedRow2[REGISTRATION_NUMBER_KEY] = '';

const expectedRows = [expectedRow1, expectedRow2];

expect(generatedRows).toEqual(expectedRows);
```

If you additionally set the name and/or email attributes of `courseUser1` to some values containing spaces at the start or end, then you could also test that the values are trimmed when exporting.